group-aws/utility-modules.md

```diff
@@ -0,0 +1,104 @@
+		get_rds_instance() -> rds_camel_dictionary
+		rds_instance_to_facts(camel_dict) -> rds_facts_snake_dict()
+		get_rds_snapshot() -> snapshot_camel_dictionary
+		rds_snapshot_to_facts(camel_dict) -> rds_facts_snake_dict()
+        resource.py - functions related to any AWS resources
+            def wait_for_status(resource, status)
+        vpc.py
+
+Please note that various functions not listed above which are included in AnsibleModule may not be
+replicated by AnsibleAWSModule.  This is deliberate to allow future simplification of the base
+class of AnsibleAWSModule.  If you have a good reason for making other functions available please
+discuss this with the AWS working group who should be happy to help or at least explain why the
+function isn't available.
+
+## Using the module utilities
+
+*This section is temporary until the module utilities become standard and should be moved to the*
+*[AWS module guildelines in the amazon modulesdirectory](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/GUIDELINES.md)*
+*once accepted*
+
+Currently the module utilities directory is should be considered preview code.  New modules which
+are to be included in Ansible should use the code.  Modules which are published outside the main
+ansible repository should _not_ include this code before ansible 2.5
+
+    from ansible.module_utils.aws.core import AnsibleAWSModule
+
+    argument_spec = dict(
+        name=dict(required=True),
+        state=dict(default='present', choices=['present', 'absent']),
+    )
+
+    module = AnsibleAWSModule(argument_spec=argument_spec,
+                              supports_check_mode=True,
+                              mutually_exclusive=mutually_exclusive,
+                              required_together=required_together,
+                              required_if=required_if)
+
+    check_mode = module.check_mode
+
+    try:
+        client = module.connect(resource='lambda') 
+    except Exception as e:
+        module.fail_json_aws(e, msg="Trying to connect to AWS")
+
+    # do other things using the client we now have. 
+
+Note that, as an "exception" to ansible normal practices, catching `Exception` should be acceptable
+here because `fail_json_aws()` should be able to deal with all boto derived exceptions.
+
+In the case where fail_json_aws() gets a non AWS exception it will record a traceback attempt to
+convert it to a string and incude that string in the final message.  In most cases this will give a
+nice printout.  Beware, though, that there are no rules for python exceptions so if your exceptions
+are abnormal (e.g. you need to call the .beam_me_up() message before you can get a string) you
+should verify that it works correctly expecially if your exception includes a `response` member.
+Please contact the AWS working group with any problematic examples you find.
```

Comment:
I'm not sure catching `Exception` is ok in this case, since if a different exception is raised (say, `KeyError`) there's never a `.response` attribute leading to an obscure failure.